torch/testing/_internal/common_methods_invocations.py

```diff
@@ -692,6 +718,7 @@ def __init__(self,
 
         # We run the sampling functions without tracking the gradiends of the creation of inputs
         self.sample_inputs_func = torch.no_grad()(sample_inputs_func)
+        self.reference_inputs_func = None if reference_inputs_func is None else torch.no_grad()(reference_inputs_func)
```

Comment:
Why no_grad?

Reply:
Consistency with sample_inputs_func, which is also run in a no_grad() context to prevent the tensors used in SampleInputs, which are often constructed using a few operations, do not have a grad history